CMakeLists.txt

```diff
@@ -22,11 +22,62 @@ target_link_directories(firebase INTERFACE
   third_party/firebase-development/usr/libs/windows/deps/app
   third_party/firebase-development/usr/libs/windows/deps/app/external)
 target_link_libraries(firebase INTERFACE
+  absl_bad_optional_access
```

Comment:
Chatted about this live, but this is a bit divergent to how we've modeled this in the Swift PM version of things. In the SwiftPM version of things we've decided to link these required libraries against the Firestore library itself as opposed to the underlying `firebase` library. 

We agreed that we should model things similarly across build systems.